test/unit/pdf_find_controller_spec.js

```diff
@@ -264,5 +265,327 @@ describe("pdf_find_controller", function () {
+    const { eventBus, pdfFindController } = await initPdfFindController();
+
+    await testSearch({
+      eventBus,
+      pdfFindController,
+      state: {
+        query: "trace-based  just-in-time",
+        caseSensitive: false,
+        entireWord: false,
+        phraseSearch: true,
+        findPrevious: false,
+      },
+      matchesPerPage: [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
+      selectedMatch: {
+        pageIndex: 0,
+        matchIndex: 0,
+      },
+      pageMatches: [
+        [0],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [2087],
+      ],
+      pageMatchesLength: [
+        [24],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [24],
+      ],
+    });
+  });
+
+  it("performs a search with square brackets in the query", async function () {
+    const { eventBus, pdfFindController } = await initPdfFindController();
+
+    await testSearch({
+      eventBus,
+      pdfFindController,
+      state: {
+        query: "[Programming Languages]",
+        caseSensitive: false,
+        entireWord: false,
+        phraseSearch: true,
+        findPrevious: false,
+      },
+      matchesPerPage: [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+      selectedMatch: {
+        pageIndex: 0,
+        matchIndex: 0,
+      },
+      pageMatches: [[1501]],
+      pageMatchesLength: [[25]],
+    });
+  });
+
+  it("performs a search with parenthesis in the query", async function () {
+    const { eventBus, pdfFindController } = await initPdfFindController();
+
+    await testSearch({
+      eventBus,
+      pdfFindController,
+      state: {
+        query: "\t   (checks)",
+        caseSensitive: false,
+        entireWord: false,
+        phraseSearch: true,
+        findPrevious: false,
+      },
+      matchesPerPage: [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+      selectedMatch: {
+        pageIndex: 1,
+        matchIndex: 0,
+      },
+      pageMatches: [[], [201]],
+      pageMatchesLength: [[], [9]],
+    });
+  });
+
+  it("performs a search with a final dot in the query", async function () {
+    const { eventBus, pdfFindController } = await initPdfFindController();
+
+    // The whitespace after the dot mustn't be matched.
+    const query = "complex applications.";
+
+    await testSearch({
+      eventBus,
+      pdfFindController,
+      state: {
+        query,
+        caseSensitive: false,
+        entireWord: false,
+        phraseSearch: true,
+        findPrevious: false,
+      },
+      matchesPerPage: [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+      selectedMatch: {
+        pageIndex: 0,
+        matchIndex: 0,
+      },
+      pageMatches: [[1946]],
+      pageMatchesLength: [[query.length]],
```

Comment:
Here, and elsewhere below, it feels slightly "safer" to *explicitly* specify the expected values since that reduces the risk of any future changes accidentally breaking something.